modules/configuration/php/configuration.class.inc

```diff
@@ -47,22 +47,22 @@ class Configuration extends \NDB_Form
     function setup()
     {
         parent::setup();
-        $config =& \NDB_Config::singleton();
-        $DB     =& \Database::singleton();
+        $config = \NDB_Config::singleton();
+        $DB     = \Database::singleton();
 
-        $scans = $DB->pselect(
+        $scans      = $DB->pselect(
             "SELECT Scan_type
              FROM mri_scan_type",
             array()
         );
+        $scan_types = array('' => '');
         foreach ($scans as $type) {
             $val = $type['Scan_type'];
-            $scan_types[$val] =$val;
+            $scan_types[$val] = $val;
         }
-        $scan_types = array_merge(array('' => ''), $scan_types);
 
-        $instruments = \Utility::getAllInstruments();
-        $instruments = array_merge(array('' => ''), $instruments);
+        $instruments     = \Utility::getAllInstruments();
+        $instruments[''] = '';
```

Comment:
??

Reply:
If you look at the previous line 65, that's what it was doing, creating an array element with both the key and value set to an empty string.

I rewrote it just to avoid the function call and make it more clear.

Reply:
Yes, I do understand that but do you know why declare such an empty key-value pair?

Reply:
I don't know why but I didn't want to alter the functionality.

Reply:
Sound good.
Should check why tpl_data['instruments'] need some empty entry. Possibly some poor underlying design.